src/Compiler/CompilerOptions.cpp

```diff
@@ -205,6 +205,57 @@ std::string getTargetCPUOption() {
   return (mcpu != "") ? "--mcpu=" + mcpu : "";
 }
 
+// Accel.
+static accel::Accelerator::Kind getAccelKindFromString(
+    const std::string &str, bool &found) {
+  found = true;
+  // Test each existing accelerator, returning its Kind when found.
+  APPLY_TO_ACCELERATORS(ACCEL_CL_ENUM_FROM_STRING, str);
+  // No specific accelerator found, check if we have Kind::NONE
+  if (str.compare(std::string("NONE")) == 0)
+    return accel::Accelerator::Kind::NONE;
+  // No valid accelerator found, report failure.
+  found = false;
+  return accel::Accelerator::Kind::NONE;
+}
+
+// Return 0 on success, nonzero on error.
```

Comment:
Could this function return a `bool` instead to indicate success/failure ?

Reply:
I agree in principle but its a bit complicated because it is implemented in the `  APPLY_TO_ACCELERATORS(ACCEL_CL_ENUM_FROM_STRING, str);` macro. 

If you really want it, I can add an additional variable, the macro can set it, and then return true...

Reply:
ok if it adds more complication then is not worth it. Is fine as is.

Reply:
Actually, the code is much more succinct with your suggestion, implemented it.